### ISSUE TITLE
problem: currency pages are not indexable

### ISSUE THREAD

**Issue by AshHimself** [Blockrazor/blockrazor#2071]
problem: currency pages are not indexable

Solution: make them indexable with some basic SEO.

**Comment by anbud** [comment]
@emurgobot claim

**Comment by anbud** [comment]
I actually found a better solution than the previous one. By using [puppeteer](https://github.com/GoogleChrome/puppeteer) we can render some SEO critical pages on the server and serve pure HTML to crawlers. Puppeteer is just a headless Chrome browser, so it renders pages perfectly fine. 

I've implemented this approach on a couple of pages in #2072. Adding new pages is as easy as adding `SeoRouter.route` calls, but we shouldn't add too much of them (not all pages are SEO critical) as server rendering with puppeteer can be a little bit resource intensive. To kinda mitigate that, I've implemented a simple caching mechanism that stores generated HTML pages in a mongo collection. If it becomes a bottleneck, we can easily switch it to something more advanced, like Redis caching.

@emurgobot done
